spring-ws-security/src/test/java/org/springframework/ws/soap/security/xwss/callback/SpringDigestPasswordValidationCallbackHandlerTest.java

```diff
@@ -47,6 +48,15 @@
 
 	private PasswordValidationCallback callback;
 
+	/**
+	 * add a tearDown method at the beginning, 
+	 * in case {@code SecurityContextHolder} not cleaned before this class
+	 */
+	@BeforeAll
+	public static void beforeClassTearDown() { 
+		SecurityContextHolder.clearContext();
+	}
+	
 	@BeforeEach
```

Comment:
If we put a clearContext in the `@BeforeEach` of pertinent test classes, we can reduce the complexity of this implementation and make it clear what is happening.

Reply:
Similar to the previous test class, I edited the code and moved the `clearContext ()` method into `setup()` to make it more simple than before.
My original purpose is to call the `clearContext ()` only in `@BeforeAll`, since we already clear the context in `@AfterEach`. But in order to make it simple, move it into setup() method is more decent.
Thanks for your suggestion!